lib/commands/moveUnlockedJobsToWait-4.lua

```diff
@@ -0,0 +1,54 @@
+--[[ 
+  Looks for unlocked jobs in the active queue. There are two circumstances in which a job
+   would be in 'active' but NOT have a job lock:
+   
+     Case A) The job was being worked on, but the worker process died and it failed to renew the lock.
+       We call these jobs 'stalled'. This is the most common case. We resolve these by moving them
+       back to wait to be re-processed. To prevent jobs from cycling endlessly between active and wait,
+       (e.g. if the job handler keeps crashing), we limit the number stalled job recoveries to MAX_STALLED_JOB_COUNT.
+
+    DEPRECATED CASE:
```

Comment:
I will remove the comment. With single instance redis, this case could not happen anymore since the lock is obtained atomically with the move. In the future, if we want to support redis with replication, we need to re-introduce redlock, but we will be able to take the lock *before* moving the job, so in practice the case should not happen either.